app/services/port.js

```diff
@@ -9,53 +10,49 @@ export default Service.extend(Evented, {
     this._super(...arguments);
 
     /*
-     * An array of objects of the form:
-     * { applicationId, applicationName }
+     * A dictionary of the form:
+     * { applicationId: applicationName }
      */
-    this.detectedApplications = [];
+    this.detectedApplications = {};
+    this.applicationId = undefined;
+    this.applicationName = undefined;
 
     this.adapter.onMessageReceived(message => {
       if (message.type === 'apps-loaded') {
-        message.apps.forEach(app => {
-          if (!this.detectedApplications.mapBy('applicationId').includes(app.applicationId)) {
-            this.detectedApplications.pushObject(app);
-          }
-        });
-      }
-    });
-
-    this.adapter.onMessageReceived(message => {
-      const { applicationId, applicationName } = message;
-
-      if (message.type === 'app-list') {
-        const apps = JSON.parse(message.appList);
-        apps.forEach((app) => {
-          if (!this.detectedApplications.mapBy('applicationId').includes(app.applicationId)) {
-            this.detectedApplications.push(app);
-          }
+        message.apps.forEach(({ applicationId, applicationName }) => {
```

Comment:
Does this not need to do a `push` into `this.detectedApplications`? Is it not a list?

Reply:
It is not a list anymore, it is a dictionary/object of `{ applicationId: applicationName }` (e.g. `{ "ember123": "Ember Inspector", "ember456": "Ember Observer" }`), so it uses `Ember.set` to instead

Reply:
Ah, I see. So this is setting the application name under the key of application id. Makes sense 👍 .